bazel/foreign_cc/curl.patch

```diff
@@ -0,0 +1,26 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
```

Comment:
upstream this patch?

Reply:
We will present it. There is the question of which each change had intended to do, so need to walk each defect up the commit history to present the correction without breaking other users' expectations. In light of the fact that curl disappears per https://github.com/envoyproxy/envoy/issues/11816 sometime in the future, acceptance of this correction upstream, as-is or further modified, shouldn't be a consideration to getting this PR merge to fix our build with our expectations.

Reply:
See https://github.com/curl/curl/pull/6050